src/main/scala/com/databricks/spark/csv/util/TypeCast.scala

```diff
@@ -35,8 +35,9 @@ object TypeCast {
    * @param datum string value
    * @param castType SparkSQL type
    */
-  private[csv] def castTo(datum: String, castType: DataType, nullable: Boolean = true): Any = {
-    if (datum == "" && nullable && !castType.isInstanceOf[StringType]){
+  private[csv] def castTo(datum: String, castType: DataType, nullable: Boolean = true,
```

Comment:
Here, I think the binary incompatibility problem is that we don't have something like Spark's MiMa exclude generator to handle the `private[csv]` excludes.